Fix click area to select hunk on both sides of diff

jwbth opened the pull request:
<!--
What GitHub Desktop issue does this PR address? (for example, #1234)
-->

Closes #13325

## Description

Add a `z-index` to elements responsible for selecting a hunk on the both sides of a diff, so that the click area is above the row and not obstructed by its right-side part.

### Screenshots

<!--
If this PR touches the UI layer of the app, please include screenshots or animated gifs to show the changes.
-->
Restored click area:
![image](https://user-images.githubusercontent.com/33615628/141791882-b47f2cda-4d89-449e-ad5e-58040648941f.png)

### Gifs
Before: 
![w7EtEmWaCW](https://user-images.githubusercontent.com/33615628/141791717-db552345-b914-41ed-a8d8-96c9d1602340.gif)
After:
![tT91g5D0vv](https://user-images.githubusercontent.com/33615628/141791731-ac5c1214-0c42-48d4-a41e-4a1a2c913899.gif)

## Release notes

<!--
You can leave this blank if you're not sure.
If you don't believe this PR needs to be mentioned in the release notes, write "Notes: no-notes".
-->

Notes:

jwbth committed:
Fix click area to select hunk on both sides of diff (#13325)

sergiou87 commented on app/styles/ui/_side-by-side-diff.scss:
```diff
@@ -138,6 +138,7 @@
     width: 20px;
     left: calc(50% - 10px);
     top: 0;
+    z-index: 1;
```
We'd like to avoid setting the `z-index` because it could end up leading to complex situations where we have to assign random z-index number to different elements to make everything work as expected.

Instead, I've looked into this particular bug, and it seems we can fix it by removing this line…
```suggestion
```
…and reordering the hunk handle when it's rendered. You can apply these changes to fix it in all cases:
```diff
diff --git a/app/src/ui/diff/side-by-side-diff-row.tsx b/app/src/ui/diff/side-by-side-diff-row.tsx
index f1decce05..bd94942b4 100644
--- a/app/src/ui/diff/side-by-side-diff-row.tsx
+++ b/app/src/ui/diff/side-by-side-diff-row.tsx
@@ -204,12 +204,12 @@ export class SideBySideDiffRow extends React.Component<
               {this.renderContentFromString('')}
               {this.renderWhitespaceHintPopover(DiffColumn.Before)}
             </div>
-            {this.renderHunkHandle()}
             <div className="after">
               {this.renderLineNumber(lineNumber, isSelected)}
               {this.renderContent(row.data)}
               {this.renderWhitespaceHintPopover(DiffColumn.After)}
             </div>
+            {this.renderHunkHandle()}
           </div>
         )
       }
@@ -241,12 +241,12 @@ export class SideBySideDiffRow extends React.Component<
               {this.renderContent(row.data)}
               {this.renderWhitespaceHintPopover(DiffColumn.Before)}
             </div>
-            {this.renderHunkHandle()}
             <div className="after">
               {this.renderLineNumber()}
               {this.renderContentFromString('')}
               {this.renderWhitespaceHintPopover(DiffColumn.After)}
             </div>
+            {this.renderHunkHandle()}
           </div>
         )
       }
@@ -259,12 +259,12 @@ export class SideBySideDiffRow extends React.Component<
               {this.renderContent(before)}
               {this.renderWhitespaceHintPopover(DiffColumn.Before)}
             </div>
-            {this.renderHunkHandle()}
             <div className="after" onMouseEnter={this.onMouseEnterLineNumber}>
               {this.renderLineNumber(after.lineNumber, after.isSelected)}
               {this.renderContent(after)}
               {this.renderWhitespaceHintPopover(DiffColumn.After)}
             </div>
+            {this.renderHunkHandle()}
           </div>
         )
       }
```

One of the changes affect added lines, which already worked… except for this annoying glitch:
![Kapture 2021-11-16 at 13 24 56](https://user-images.githubusercontent.com/1083228/141984955-d10e90b5-5b60-4db0-9a2d-f9dfb6aa051d.gif)

With the diff above, it will keep the same cursor over the area of the whole hunk handle 😄

jwbth replied:
Thanks, that was the other possible solution that I was thinking. I've applied the suggested changes 🙂

sergiou87 reviewed (changes requested):
Nice catch @jwbth !! Thank you very much for reporting and fixing this 😄 

However, we think there is a better fix for this bug 🤞  I hope you don't mind applying the suggested changes.

Thanks for your patience!

jwbth committed:
Revert app/styles/ui/_side-by-side-diff.scss

Co-authored-by: Sergio Padrino <sergiou87@github.com>

jwbth committed:
Reorder diff row elements in side by side diff

These changes are suggested by @sergiou87 at https://github.com/desktop/desktop/pull/13326#pullrequestreview-807231206.

sergiou87 reviewed (approved):
Thank you very much for your contribution!!! ❤️
